libnetwork/drivers/bridge/port_mapping_linux.go

```diff
@@ -134,6 +146,42 @@ func (n *bridgeNetwork) addPortMappings(
 		}
 	}
 
+	// Now the iptables rules are set up, it's safe to start the userland proxy.
+	// (If it was started before the iptables rules were created, it may have
+	// accepted a connection, then become unreachable due to NAT rules sending
+	// packets directly to the container.)
+	// If not starting the proxy, nothing will ever accept a connection on the
+	// socket. But, listen anyway so that the binding shows up in "netstat -at".
+	somaxconn := 0
+	if proxyPath != "" {
+		somaxconn = getSOMAXCONN()
```

Comment:
```suggestion
		somaxconn = -1
```
[The kernel will silently cap the backlog to SOMAXCONN for you.](https://man7.org/linux/man-pages/man2/listen.2.html#:~:text=If%20the%20backlog,the%20value%20128.) [(Kernel source)](https://elixir.bootlin.com/linux/v6.9.8/source/net/socket.c#L1876)

Reply:
Oh, ok - I'll change it. I wonder why `net.ListenTCP` ends up [reading the config](https://github.com/golang/go/blob/87ec2c959c73e62bfae230ef7efca11ec2a90804/src/net/sock_linux.go#L34-L54).